cloud/cluster/cluster.go

```diff
@@ -5,121 +5,133 @@ package cluster
 
 import (
 	"sort"
+	"sync"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/twitter/scoot/common/stats"
 )
 
-// Cluster represents a group of Nodes and has mechanisms for receiving updates.
-type Cluster struct {
-	state    *state
-	reqCh    chan interface{}
-	updateCh chan ClusterUpdate
-	subs     []chan []NodeUpdate
+var ClusterUpdateLoopFrequency time.Duration = time.Duration(250) * time.Millisecond
+
+// Cluster represents a group of Nodes and has mechanism for
+// setting current Node list and receiving correlating Node updates.
+type Cluster interface {
+	RetrieveCurrentNodeUpdates() []NodeUpdate // used by scheduler's cluster state to get updates
```

Comment:
nit: iface comments shouldn't rely on external things (scheduler, fetcher) for usage context